v2/routing.go

```diff
@@ -132,14 +144,29 @@ func (d *DHT) PutValue(ctx context.Context, key string, value []byte, option ...
 		return fmt.Errorf("store record locally: %w", err)
 	}
 
-	// TODO reach out to Zikade
-	panic("implement me")
+	return nil
 }
 
 func (d *DHT) GetValue(ctx context.Context, key string, option ...routing.Option) ([]byte, error) {
 	ctx, span := d.tele.Tracer.Start(ctx, "DHT.GetValue")
 	defer span.End()
 
+	v, err := d.GetValueLocal(ctx, key)
+	if err != nil {
+		return v, nil
+	}
+	if !errors.Is(err, ds.ErrNotFound) {
+		return nil, fmt.Errorf("put value locally: %w", err)
+	}
+
+	panic("implement me")
+}
+
+// GetValueLocal retrieves a value from the local datastore without querying the network.
+func (d *DHT) GetValueLocal(ctx context.Context, key string) ([]byte, error) {
```

Comment:
rather unexport and only export when it's needed to decrease API surface?

Reply:
:+1: